### ISSUE TITLE
Publishing @types/ ?

### ISSUE THREAD

**Issue by KyleWiteck** [transitive-bullshit/create-react-library#257]
Is the option available for publishing the typescript declarations @types/ module?

I have tried to find a way for awhile, having little to no luck. Thanks.

**Comment by psyycker** [comment]
All the code of the CLI is written in plain JS. Why would you need typescript support for a CLI?
Tell me if I get it wrong

**Comment by KyleWiteck** [comment]
I guess I asked this wrong to be honest. Was wondering is there was a way to build a typescript module all together with their types?

We always use typescript within our organization and host modules through azure artifacts. Having a typescript build would allow us to use the module with intellisense (which speeds up production knowing what is allowed in the modules components prop for example) and just make for a better experience for juniors and prevent a bunch of breaks in our apps.

Please let me know if this is reasonable for you guys. If not I will have to build from scratch. Thanks!

**Comment by psyycker** [comment]
Oh I see what you mean. Well this is something to change with the build process. CRL gives you a default template for your favourite typing. But then, free to you to change your tsconfig.json to adapt what you'd need!
I don't know typescript enough to help you, but I'm pretty sure there is an option for yours tsconfig.json to have what you'd need.

**Comment by yoeran** [comment]
Just started my first library with CRL and it provided me with correct types just fine initially. However, I think I've changed something somewhere in a config file which caused the typings to no longer be included correctly. For me the solution was to add `"typings": "dist/src/index.d.ts"` to the `package.json`. As I looked in the `dist` folder that happened to be the new (?) location all of a sudden. Maybe that helps!
